.github/workflows/build-strict.yml

```diff
@@ -45,5 +41,73 @@ jobs:
           rm -rf .lake/packages/proofwidgets/.lake/build/ir
           lake build ProofWidgets
 
+      - name: confirm all modules are included
+        run: lake exe mk_all --check
+
       - name: build project
         run: lake build --wfail
+
+      - name: Build blueprint and copy to `docs/blueprint`
+        uses: xu-cheng/texlive-action@v2
+        with:
+          docker_image: ghcr.io/xu-cheng/texlive-full:20231201
+          run: |
+            apk update
+            apk add --update make py3-pip git pkgconfig graphviz graphviz-dev gcc musl-dev
+            git config --global --add safe.directory $GITHUB_WORKSPACE
+            git config --global --add safe.directory `pwd`
+            python3 -m venv env
+            source env/bin/activate
+            pip install --upgrade pip requests wheel
+            pip install pygraphviz --global-option=build_ext --global-option="-L/usr/lib/graphviz/" --global-option="-R/usr/lib/graphviz/"
+            pip install leanblueprint
+            leanblueprint pdf
+            mkdir docs
+            cp blueprint/print/print.pdf docs/blueprint.pdf
+            leanblueprint web
+            cp -r blueprint/web docs/blueprint
+
+      - name: check blueprint declarations
+        run: lake exe checkdecls blueprint/lean_decls
+
+      - name: cache mathlib docs
+        uses: actions/cache@v4
+        with:
+          path: |
+            .lake/build/doc/init
+            .lake/build/doc/lake
+            .lake/build/doc/lean
+            .lake/build/doc/std
+            .lake/build/doc/mathlib
+            .lake/build/doc/declarations
+            !.lake/build/doc/declarations/declaration-data-algebrainlean*
+          key: mathlibdoc-${{ hashfiles('lake-manifest.json') }}
+          restore-keys: |
+            mathlibdoc-
+
+      - name: build documentation
+        run: lake -R -Kenv=dev build AlgebraInLean:docs
+
+      - name: move documentation to `docs/docs`
+        run: |
+          sudo chown -R runner docs
+          cp -r .lake/build/doc docs/docs
+
+      - name: make sure the cache works
+        run: |
+          mv docs/docs .lake/build/doc
```

Comment:
What is this step even really doing? Do we use `.lake/build` for anything after this step?

Reply:
Also, wouldn't this prevent the docs from being uploaded to GH pages?